packages/vite/src/node/__tests__/plugins/define.spec.ts

```diff
@@ -37,4 +37,22 @@ describe('definePlugin', () => {
       'const isSSR = false;'
     )
   })
+
+  test('ignores import\0.meta.env inside strings', async () => {
```

Comment:
Vitest is replacing `import.meta.env` with `process.env` in unit tests, which coincidentally is the bug I'm trying to fix (so *meta* 🤣). I'm wondering if it's actually the same bug.

I inserted `\0` in this test's strings to prevent that.

Reply:
This is such a deep level of programming irony. Got me laughing.

Reply:
I think this is because this PR doesn't strip literals in SSR [here](https://github.com/vitejs/vite/blob/01fd595dcd80d06526744913b606d90bbe86f9b2/packages/vite/src/node/plugins/define.ts#L129-L131), which we may want to do as well. Otherwise we're using Vitest linked to local Vite, so theoretically this line should be fixed too.

Since we're already parsing in SSR dev (though I heard Anthony is experimenting something), another parse/tokenize here maybe isn't too bad, unless we could share a cache via `this.parse` perhaps.